### PR TITLE
fix(node-ws): `CloseEvent` is not defined

### LINKED ISSUE #595: [@hono/node-ws] ReferenceError: CloseEvent is not defined
I'm using Node `v20.14.0` with @hono/node-ws `v1.0.1` and when listening to the onClose event it throws the following error:
```js
events.onClose?.(new CloseEvent("close"), ctx);
                     ^

ReferenceError: CloseEvent is not defined
```

### PR DESCRIPTION
**Pull request by Yovach**
Currently, `CloseEvent` doesn't exist in the latest versions of Node.js (v18.20.4, v20.15.1 and v22.4.) and you'll get the following error `Uncaught ReferenceError: CloseEvent is not defined` if you have this :

```javascript
app.get(
  '/',
  upgradeWebSocket(() => ({
    onClose() {
      // doing some stuff
    },
  }))
)
```

Fixes #595

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @Yovach 

Looks good to me! One thing. Can you run `yarn changeset` on the top of the project and push the changeset?

@nakasyou Can you also review this?

File: packages/node-ws/src/events.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+interface CloseEventInit extends EventInit {
+    code?: number;
+    reason?: string;
+    wasClean?: boolean;
+}
+
+/**
+ * @link https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent
+ */
+export class CloseEvent extends Event {
```
Comment (nakasyou) [inline packages/node-ws/src/events.ts:10]:
```suggestion
export const CloseEvent = globalThis.CloseEvent ?? class extends Event {
```
What do you think about that?
If `CloseEvent` defined, the middleware should use it. This is effective for runtimes where `CloseEvent` is compatible with Node.js is defined.

**Review by nakasyou — CHANGES_REQUESTED** [review]

File: packages/node-ws/src/events.ts (hunk 1 above)
Comment (yusukebe) [reply packages/node-ws/src/events.ts:10]: Seems to be good.

If we do so, we may have to have the CI on both Node.js, which has `CloseEvent` and does not.

Comment (Yovach) [reply packages/node-ws/src/events.ts]: > What do you think about that? If `CloseEvent` defined, the middleware should use it. This is effective for runtimes where `CloseEvent` is compatible with Node.js is defined.

Nice idea, I added your code to the `events.ts`. Thanks for your suggestion!

Comment (yusukebe) [reply packages/node-ws/src/events.ts]: > Currently, `CloseEvent` is not exposed to `globalThis` in Node.js 22.5.0 so I think we don't need to have multiple Node.js versions in the CI for now.

Okay! Let's go with it.

**Comment by yusukebe** [comment]
@nakasyou Thank you for your review!

@Yovach I'll merge it into the main and release the new version. Thank you!

**Pull request merged**